TrkFilters/src/HelixFilter_module.cc

```diff
@@ -191,6 +201,61 @@ namespace mu2e
     }
     return true;
   }
+
+
+  float   HelixFilter::helixHitRatio(const HelixSeed* helix){
+
+    int      nCH = helix->hits().size();
+    int      nFaces(36); //was 36, now doing stations instead of faces
+    float    expected_hits(0);
+    float    z_step(178.); //are they really equally spaced? was 89 before
+    float    minR_tracker(380.);
+    float    maxR_tracker(860.);
+
+    //find the z of the first hit (closest to the stopping target)
+    const ComboHit*    hit(0);
+    double   firstHitZ(1e5);
+    double   lastHitZ(-1e5);
+    
+    for (int k=0; k<nCH; ++k){
+      hit   = &helix->hits().at(k);
+      if (hit->pos().z() > lastHitZ){
+	lastHitZ = hit->pos().z();
+      }
+    }
+
+    for (int i=0; i<nCH; ++i){
+      hit   = &helix->hits().at(i);
+      if (hit->pos().z() < firstHitZ){
+	firstHitZ = hit->pos().z();
+      }
+    }
+
+    for (int i=0; i<nFaces;i++){
+      //float   z = zStart_tracker + i*z_step;
+      float   z = firstHitZ + (double)i*z_step;
+      if (z < firstHitZ )  continue;
+      if (z > lastHitZ  )  continue;
+
+      XYZVec  pos;
+      pos.SetZ(z);
+      helix->helix().position(pos);
+      
+      //now check that we are in the active area of tracker
+      double   hitR = sqrt(pos.x()*pos.x() + pos.y()*pos.y());
+      if (hitR < minR_tracker)  continue;
+      if (hitR > maxR_tracker)  continue;
+
+      ++expected_hits;
+    }
+
+    //each face has two planes. We are assuming 2 ComboHits per face
+    float   hitRatio = nCH / (2.*expected_hits);
```

Comment:
This assumes every plane produces 2 hits, which may be 'true' statistically, but  It would be more consistent if you compared to the number of planes with hits, as expected_hits is really expected_planes.  In any event the value 2 is an algorithmic parameter and should be in fcl.

Reply:
Thank you for the suggestion. I'm now testing the plane counting instead oh the comboHits.